src/main/flight/imu.c

```diff
@@ -277,6 +290,9 @@ static void imuMahonyAHRSupdate(float dt, const fpVector3_t * gyroBF, const fpVe
             // Ignore magnetic inclination
             vMag.z = 0.0f;
 
+            // Apply magnetic declination correction
+            rotationMatrixRotateVector(&vMag, &vMag, &magDeclinationRMat);
```

Comment:
It may be easier (and actually cleaner) to use true north vector ( `[cos(declination, sin(declination), 0]` (?) ) as reference

Reply:
@ledvinap you're a genius! It didn't occur to me that I can just figure our true North vector from declination 👍